modules/effects/arrow-effects/src/main/kotlin/arrow/effects/typeclasses/Concurrent.kt

```diff
@@ -0,0 +1,82 @@
+package arrow.effects.typeclasses
+
+import arrow.Kind
+import arrow.core.*
+import kotlin.coroutines.CoroutineContext
+
+interface Concurrent<F> : Async<F> {
+
+  fun <A> Kind<F, A>.startF(ctx: CoroutineContext): Kind<F, Fiber<F, A>>
+
+  fun <A, B> racePair(ctx: CoroutineContext, fa: Kind<F, A>, fb: Kind<F, B>): Kind<F, Either<Tuple2<A, Fiber<F, B>>, Tuple2<Fiber<F, A>, B>>>
+
+  //TODO blocked by Async#asyncF (https://github.com/arrow-kt/arrow/issues/1124)
+  //fun <A> cancelable(cb: ((Either<Throwable, A>) -> Unit) -> Kind<F, Unit>): Kind<F, A> =
+
+  fun <A, B> raceN(ctx: CoroutineContext, fa: Kind<F, A>, fb: Kind<F, B>): Kind<F, Either<A, B>> =
+    racePair(ctx, fa, fb).flatMap {
+      it.fold({ (a, b) ->
+        b.cancel.map { a.left() }
+      }, { (a, b) ->
+        a.cancel.map { b.right() }
+      })
+    }
+
+  fun <A, B, C> raceN(ctx: CoroutineContext, fa: Kind<F, A>, fb: Kind<F, B>, fc: Kind<F, C>): Kind<F, Either<A, Either<B, C>>> =
```

Comment:
_eye twitch_ is this nested Either how it's presented in Scala?

Reply:
Probably not. They have proper `Product`s _sigh_

Reply:
@pakoito this is how it's currently presented on [IO.Companion](arrow-effects-instances) and is why I choose it. I am not a huge fan of this tho.

Reply:
@pakoito I made an attempt to make this nicer. Check the aliases at the bottom of the file. Thoughts?